src/Dev/State/FixtureTestState.php

```diff
@@ -0,0 +1,229 @@
+     * @param SapphireTest $test
+     */
+    public function tearDown(SapphireTest $test)
+    {
+        if ($this->testNeedsDB($test)) {
+            $test::tempDB()->rollbackTransaction();
+        }
+    }
+
+    /**
+     * Called once on setup
+     *
+     * @param string $class Class being setup
+     */
+    public function setUpOnce($class)
+    {
+        $this->fixtureFactories[strtolower($class)] = Injector::inst()->create(FixtureFactory::class);
+    }
+
+    /**
+     * Called once on tear down
+     *
+     * @param string $class Class being torn down
+     */
+    public function tearDownOnce($class)
+    {
+        unset($this->fixtureFactories[strtolower($class)]);
+        $class::tempDB()->clearAllData();
+    }
+
+    /**
+     * @param string $class
+     *
+     * @return bool|FixtureFactory
+     */
+    public function getFixtureFactory($class)
+    {
+        $testClass = strtolower($class);
+        if (array_key_exists($testClass, $this->fixtureFactories)) {
+            return $this->fixtureFactories[$testClass];
+        }
+        return false;
+    }
+
+    /**
+     * @param FixtureFactory $factory
+     * @param string $class
+     */
+    public function setFixtureFactory(FixtureFactory $factory, $class)
+    {
+        $this->fixtureFactories[strtolower($class)] = $factory;
+    }
+
+    /**
+     * @param array $fixtures
+     *
+     * @param SapphireTest $test
+     *
+     * @return array
+     */
+    protected function getFixturePaths($fixtures, SapphireTest $test)
+    {
+        return array_map(function ($fixtureFilePath) use ($test) {
+            return $this->resolveFixturePath($fixtureFilePath, $test);
+        }, $fixtures);
+    }
+
+    /**
+     * @param SapphireTest $test
+     */
+    protected function loadFixtures(SapphireTest $test)
+    {
+        $fixtures = $test::get_fixture_file();
+        $fixtures = is_array($fixtures) ? $fixtures : [$fixtures];
+        $paths = $this->getFixturePaths($fixtures, $test);
+        foreach ($paths as $fixtureFile) {
+            $this->loadFixture($fixtureFile, $test);
+        }
+    }
+
+    /**
+     * @param string $fixtureFile
+     * @param SapphireTest $test
+     */
+    protected function loadFixture($fixtureFile, SapphireTest $test)
+    {
+        /** @var YamlFixture $fixture */
+        $fixture = Injector::inst()->create(YamlFixture::class, $fixtureFile);
+        $fixture->writeInto($this->getFixtureFactory(get_class($test)));
+    }
+
+    /**
+     * Map a fixture path to a physical file
+     *
+     * @param string $fixtureFilePath
+     * @param SapphireTest $test
+     *
+     * @return string
+     */
+    protected function resolveFixturePath($fixtureFilePath, SapphireTest $test)
+    {
+        // Support fixture paths relative to the test class, rather than relative to webroot
+        // String checking is faster than file_exists() calls.
+        $isRelativeToFile
+            = (strpos('/', $fixtureFilePath) === false)
```

Comment:
`int strpos ( string $haystack , mixed $needle [, int $offset = 0 ] )`
Around the wrong way?

Reply:
Ah yes.

I also like strstr() for doing simple is_in_string :P

Reply:
Hah, this has been lifted directly from `SapphireTest` - so fixing this may have unintended consequences...

I've pushed up a fix, though